### ISSUE TITLE
the type of super() should be instance, not void

### ISSUE THREAD

**Issue by LongTengDao** [microsoft/TypeScript#37847]
**TypeScript Version:**  3.8.3

**Search Terms:**

**Code**

```ts
class MyArray extends Array {
  constructor () {
    return super();
  }
}
```

**Expected behavior:** no error

**Actual behavior:** `TS2322`

**Playground Link:**

**Related Issues:**

**Comment by LongTengDao** [comment]
addon:

```ts
class MyArray extends Array {
  a :1;
  constructor () {
    super().a;// better to be able to warn not init yet
    this.a = 1;
    this.a;// ok now
  }
}
```

**Comment by 00ff0000red** [comment]
Are there any plans on fixing (investigating) this any time soon?

**Comment by cangSDARM** [comment]
any news or update?

Recently, tsc compiled `super` to an inline anonymous function, I wonder why

![image](https://github.com/microsoft/TypeScript/assets/20441896/6de8e353-aa86-46c1-af1a-3d012ab1f9c2)

**Comment by ExE-Boss** [comment]
@cangSDARM
Can you post the source code which reproduces that bug with the relevant `tsconfig.json` configuration?
````md
```ts repro
// source code here
```

[Workbench Repro](https://www.typescriptlang.org/dev/bug-workbench/)
````

**Comment by cangSDARM** [comment]
> @cangSDARM Can you post the source code which reproduces that bug with the relevant `tsconfig.json` configuration?
> 
> ```
> ```ts repro
> // source code here
> ```
> 
> [Workbench Repro](https://www.typescriptlang.org/dev/bug-workbench/)
> ```

Sry, my mistake. I did a little more digging, and this is esbuild issue :) https://github.com/evanw/esbuild/issues/3538
